Tests/WitsmlExplorer.Api.Tests/Workers/LogWorkerToolsTests.cs

```diff
@@ -0,0 +1,189 @@
+            loggerFactory.AddSerilog(Log.Logger);
+            ILogger<CreateObjectOnWellboreJob> logger = loggerFactory.CreateLogger<CreateObjectOnWellboreJob>();
+            _worker = new CreateObjectOnWellboreWorker(logger, witsmlClientProvider.Object);
+        }
+
+        [Fact]
+        public async Task GetLogTest_OK()
+        {
+            WitsmlLogs returnedWitsmlLog = ReturnedWitsmlLog();
+            CreateObjectOnWellboreJob job = CreateJobTemplate(WitsmlLog.WITSML_INDEX_TYPE_MD);
+            List<WitsmlLogs> createdLogs = new();
+            _witsmlClient.Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.Is<OptionsIn>((ops) => ops.ReturnElements == ReturnElements.HeaderOnly))).ReturnsAsync(returnedWitsmlLog);
+
+            var log = await LogWorkerTools.GetLog(_witsmlClient.Object, job.Object, ReturnElements.HeaderOnly);
+            Assert.Equal("51bb71c2-5e6f-4e15-ae5f-0fbc866bdad6", log.Uid);
+            Assert.Equal("2023-04-19T00:00:09Z", log.EndDateTimeIndex);
+            Assert.Equal("2023-04-19T00:00:04Z", log.StartDateTimeIndex);
+            Assert.Equal(3, log.LogCurveInfo.Count);
+        }
+
+        [Fact]
+        public async Task GetLogDataForCurveTest_OK()
+        {
+            WitsmlLogs returnedWitsmlLog = ReturnedWitsmlLog();
+            _witsmlClient.Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.Is<OptionsIn>((ops) => ops.ReturnElements == ReturnElements.HeaderOnly))).ReturnsAsync(returnedWitsmlLog);
+            WitsmlLog log = LogUtils.GetSourceLogs(WitsmlLog.WITSML_INDEX_TYPE_MD, 123.11, 123.12, "Depth").Logs.First();
+            LogUtils.SetupGetDepthIndexed(_witsmlClient, (logs) => logs.Logs.First().StartIndex?.Value == "123.11",
+                new() { new() { Data = "123.11,1," }, new() { Data = "123.12,,2" } });
+
+            LogUtils.SetupGetDepthIndexed(_witsmlClient, (logs) => logs.Logs.First().StartIndex?.Value == "123.12",
+                new() { new() { Data = "123.12,,2" } });
+            var sourceLog = await LogWorkerTools.GetLogDataForCurve(_witsmlClient.Object, log, "mnemonic", null);
+            Assert.Equal("123.11,1,", sourceLog.Data[0].Data);
+            Assert.Equal("123.12,,2", sourceLog.Data[1].Data);
+            Assert.Equal("data", sourceLog.Data[0].TypeName);
+            Assert.Equal("data", sourceLog.Data[1].TypeName);
+            Assert.Equal("Depth,DepthBit,DepthHole", sourceLog.MnemonicList);
+            Assert.Equal("data", sourceLog.TypeName);
+            Assert.Equal("m,m,m", sourceLog.UnitList);
+        }
+
+        [Fact]
+        public void CalculateProgressBasedOnIndexTest_OK()
+        {
+            var log = CreateLog();
+            string mnemonicList = "Depth,BPOS";
+            var witmslLogData = GetTestLogData(mnemonicList);
+            var result = LogWorkerTools.CalculateProgressBasedOnIndex(log, witmslLogData);
+            Assert.Equal(1.2, result);
```

Comment:
This test is a bit confusing. Progress should be between 0 and 1, not 1.2. The problem is that the data points in _result_ exceeds the specified start and end indexes from _log_. To make the test more realistic, the data you get should be within the start and end indexes.

Reply:
fixed